Documentation/security/policy/intro.rst

```diff
@@ -14,53 +14,88 @@ Policy Enforcement Modes
+  have unrestricted network access until selected by policy. Upon being selected by
+  a policy, the endpoint permits only allowed traffic. This state is per-direction
+  and can be adjusted on a per-policy basis. :ref:`See below<policy_mode_default>` for
+  more details.
 
 always
   With always mode, policy enforcement is enabled on all endpoints even if no
   rules select specific endpoints.
 
   If you want to configure health entity to check cluster-wide connectivity when 
-  you start cilium-agent with ``enable-policy=always``, you will likely want to
+  you start cilium-agent with ``enable-policy: always``, you will likely want to
   enable communications to and from the health endpoint. See :ref:`health_endpoint`.
 
 never
   With never mode, policy enforcement is disabled on all endpoints, even if
   rules do select specific endpoints. In other words, all traffic is allowed
   from any source (on ingress) or destination (on egress).
 
-To configure the policy enforcement mode at runtime for all endpoints managed by a Cilium agent, use:
+To :ref:`configure <k8s_configuration>` the policy enforcement mode, you can adjust the Helm value
+``policyEnforcementMode`` or the corresponding configuration flag ``enable-policy``
 
-.. code-block:: shell-session
 
-    $ cilium-dbg config PolicyEnforcement={default,always,never}
+.. _policy_mode_default:
 
-If you want to configure the policy enforcement mode at start-time for a particular agent, provide the following flag when launching the Cilium
-daemon:
+Endpoint default policy
+-----------------------
 
-.. code-block:: shell-session
+By default, all egress and ingress traffic is allowed for all endpoints. When
+an endpoint is selected by a network policy, it transitions in to a default-deny
+state, where only **explicitly allowed** traffic is permitted. This state is
+per-direction:
 
-    $ cilium-agent --enable-policy={default,always,never} [...]
+* If any rule selects an :ref:`endpoint` and the rule has an ingress
+  section, the endpoint goes into default deny mode for ingress.
+* If any rule selects an :ref:`endpoint` and the rule has an egress section, the
+  endpoint goes into default deny mode for egress.
 
-Similarly, you can enable the policy enforcement mode across a Kubernetes cluster by including the parameter above in the Cilium DaemonSet.
+This means that endpoints start without any restrictions, and the first
+policy will switch the endpoint's default enforcement mode (per direction).
 
-.. code-block:: yaml
+It is possible to create policies that do not enable the default-deny mode for selected
+endpoints. The field ``EnableDefaultDeny`` configures this. Rules with ``EnableDefaultDeny``
+disabled are ignored when determining the default mode.
 
-    - name: CILIUM_ENABLE_POLICY
-      value: always
+For example, this policy causes all DNS traffic to be intercepted, but does not
+block any traffic, even if it is the first policy to apply to an endpoint.
```

Comment:
What would be a use case for this rule? It could be interesting to mention one in the docs.

Reply:
Good point, added one.